atecc608a_se.c

```diff
@@ -161,6 +161,43 @@ static psa_status_t atecc608a_export_public_key(psa_key_slot_number_t key,
     atecc608a_deinit();
     return status;
 }
+static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
+                                                psa_key_lifetime_t lifetime,
+                                                psa_key_type_t type,
+                                                psa_algorithm_t alg,
+                                                psa_key_usage_t usage,
+                                                const uint8_t *p_data,
+                                                size_t data_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+
+    /* Keys 8 to 15 can store public keys. Slots 1-7 are too small. */
+    if (key_id < 7 || key_id > 15 || data_length != ATCA_PUB_KEY_SIZE)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (!PSA_KEY_TYPE_IS_PUBLIC_KEY(type))
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* We can only do ECDSA on SHA-256 */
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    ASSERT_SUCCESS_PSA(atecc608a_init());
+
+    /* Signature format is R and S integers in big-endian format.
+     * 64 bytes for P256 curve. */
+    ASSERT_SUCCESS(atcab_write_pubkey(key_id, p_data));
```

Comment:
Do we need to strip off the `04` in the front of `p_data` before passing on to the driver?

Reply:
We can do it here, you're right.